core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -979,4 +984,7 @@ private[spark] object Executor {
   // task is fully deserialized. When possible, the TaskContext.getLocalProperty call should be
   // used instead.
   val taskDeserializationProps: ThreadLocal[Properties] = new ThreadLocal[Properties]
+
+  // Used to store executorSource, for local mode only
+  var executorSourceLocalModeOnly: ExecutorSource = null
```

Comment:
Nit: Looks like this one is never been cleaned. It would be great to avoid using a global `executorSourceLocalModeOnly` to save a state of a specific executor. Can we move this to SparkEnv so that a state of one test won't be leaked to other tests?

Reply:
Thanks @zsxwing , I'll have a look at it.